controllers/storagecluster/cephobjectstores.go

```diff
@@ -186,7 +182,7 @@ func (r *StorageClusterReconciler) newCephObjectStoreInstances(initData *ocsv1.S
 							"service.kubernetes.io/topology-mode":                "Auto",
 						},
 					},
-					Instances: gatewayInstances,
+					Instances: getCephObjectStoreGatewayInstances(initData),
 					Placement: getPlacement(initData, "rgw"),
```

Comment:
What placement is assigned to the RGW? Is there is required pod anti-affinity it could be a problem if they have too high a daemon count, but it's fine if it's just preferred anti-affinity.

Reply:
We do actually apply required pod anti-affinity when it's more than 1 rgw, I think it will create a problem if someone specifies more number of rgw daemons than the number of failure domains they have. As in ODF we know we always have minimum 3 failure domains, Should we add maxvalue validator to the filed making it possible to only enter values upto 3?

https://github.com/red-hat-storage/ocs-operator/blob/b5f02792f5c46810e84508d46134608123cd1423/controllers/storagecluster/placement.go#L64-L77

Reply:
Or may be we can modify the code to not apply the required anti affinity when the rgw count is more than the failure domains, Also the mdscount*2 is more than the failure domains.

Reply:
Updated the code so that if mds or rgw pods are more than the number of failure domains, don't add required pod anti affinity.